### DISCUSSION TITLE
Possibility to reload RemoteJWKSet without blocking jwt verification

### DISCUSSION THREAD

**Discussion by umutuzgur** [panva/jose#672] [Ideas]
Hi there,

We are trying to tackle an issue we have with lambda where the jwt verification takes more than 1s when the cache max age is reached. I was hoping to kickoff the reload cycle before the cache age is reached and do a hot swap so verify doesn't need to do a refresh. As a quick solution, could we also export `RemoteJWKSet` like `LocalJWKSet` so I can call the reload function asynchronously?

**Comment by panva** [comment]
Hi @umutuzgur.

I'm not going to export those classes, but please have a look if https://github.com/panva/jose/compare/main...disc-672 would suffice, it allows introspecting the remote resolver's state and manually triggering reload.

These methods are typed, public API, but intentionally not documented since their use is rather niche.

**Reply by umutuzgur** [reply]
@panva That is even better. I didn't suggest that since I thought it would be more work for you but you implemented that very fast 👏 . Thanks for the quick response

**Reply by panva** [reply]
https://github.com/panva/jose/releases/tag/v5.3.0
